tiledb/sm/fragment/fragment_info.h

```diff
@@ -77,15 +77,15 @@ class FragmentInfo {
   /*                API                */
   /* ********************************* */
 
-  /** Appends the info about a single fragment at the end of `fragments_`. */
-  void append(const SingleFragmentInfo& fragment);
+  /**
```

Comment:
Maybe take time to add complete doc strings with parameters and returns for this file?

Reply:
This is from previous iterations, let's address in a separate PR holistically for each header file.